pip/pip-347.md

```diff
@@ -0,0 +1,80 @@
+  broker will check if user's token has the `consume` permission of `persistent://public/default/test`, if yes, broker will retrieve the permission of all subscriptions under the namespace `public/default` and get the role list of subscription `sub`.
+    - **If the role list is empty or null, broker allow to create the subscription.** However, the permission of the subscription is still not granted to the user's token.
+    - If the role list is not empty, broker will check if the role corresponding to the user's token is in the role list. If yes, broker allow to create the subscription.
+    - If the role corresponding to the user's token is not in the role list, broker will reject the request.
+
+There is a permission hole in the current design. If a user obtain the `consume` permission of a topic in management platform, he can create infinite subscriptions without notifying the administrator.
+For example, user `jack` can obtain the `consume` permission of topic `persistent://public/default/test` and the permission of subscription `sub` in management platform.
+At this time, the role list under the namespace `public/default` is 
+```
+'sub' -> ['jack']
+```
+Then, user `jack` can create other subscriptions `sub1`, `sub2`,... for topic `persistent://public/default/test` without notifying the administrator. The role list under the namespace `public/default` remain unchanged as
+```
+'sub' -> ['jack']
+```
+The administrator can't know who is the owner of the subscriptions `sub1`, `sub2`,...
+
+What is worse, if another user `tom` obtain the permission of subscription `sub1` under the same namespace `public/default` in management platform, that result into the role list under the namespace `public/default` become to 
+``` 
+'sub' -> ['jack']
+'sub1' -> ['tom']
+```
+`jack` lose the permission of subscription `sub1` immediately. Because jack do not have the permission of subscription `sub1` really.
+
+He just uses an unregistered subscription `sub1` owned by nobody to consume the messages. Once there are any people obtain the permission of subscription `sub1`, his task consuming `sub1` will be shutdown immediately.
+
+# Motivation
+
+Such kind of permission lost problem cause us lots of problem. We need a tool to find out the owner of these unregistered subscriptions first, and then we can grant the permission of these subscriptions to the real owner, in case of the permission lost problem.
+
+# Goals
+
+## In Scope
+
+- help administrator to find out the owner of the consumer
+
+## Out of Scope
+
+Maybe we improve the permission control of subscriptions in the future to fix the permission lost problem described above.
+
+
+# Detailed Design
+
+## Design & Implementation Details
+- Add a field `role` in the consumer's stat to show the owner of this consumer.
+- Add a new conf `exposeRoleInConsumerStat` to enable or disable this feature.
```

Comment:
This is not the right place.

We could add a query parameter to ask for this piece of information.
And in case you pass this parameter then we have to verify that the user has high privileges because we cannot expose the roles to normal users.

If we put this configuration in the Broker then we will allow topic stats only to administrators

Reply:
Agree, i have updated the doc, PTAL, thanks.